add strategy paper for observability (#695)
kyma-project/community: concepts/observability-strategy/configurable-monitoring/pocs/scrape-scalability/README.md

## Goal

> The collector must support scraping metrics in the prometheus pull approach. For that the [prometheusreceiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/prometheusreceiver) needs to be used.

Reviewer: ```suggestion
The collector must support scraping metrics in the Prometheus pull approach. Thus, the [Prometheus receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/prometheusreceiver) must be used.
```
That's how they write it in their doc.

> As the instance keeps state about the scraped data, one target cannot be scraped by interchanging instances (sticky targets we could say).

Reviewer: ```suggestion
Because the instance keeps state about the scraped data, one target cannot be scraped by interchanging instances (sticky targets, we could say).
```

> What will happen if there a magnitude of scrape targets and one instance cannot scrape them all?

Reviewer: ```suggestion
What will happen if there is a large amount of scrape targets and one instance cannot scrape them all?
```

Probably only sharding can solve the problem by distributing the targets to scrape to multiple instances.

## Ideas
- Coordinator: Have an instance like the otel-collector-operator which splits the scrape config into distinct sets and assignes it collector instances.

> - Self-contained Criterias: Support configs only via service discovery and annotations, then shard the targets via dynamic criterias like namespace or node by the collector itself

Reviewer: ```suggestion
- Self-contained criteria: Support configs only with Service Discovery and annotations, and the collector itself is responsible for sharding the targets with dynamic criteria like Namespace or node.
```
I guess this is what you meant?

## Proposal

> Support only dynamic confgiuration of targets via Service Discovery and annotations. Use service discovery attributes as criteria inside every collector to shard the targets over the instances.

Reviewer: ```suggestion
Support only dynamic confgiuration of targets with Service Discovery and annotations. Use Service Discovery attributes as criteria inside every collector to shard the targets over the instances.
```

This is extremely similar to line 12?
